index.js

```diff
@@ -4,20 +4,29 @@ var crypto = require('crypto');
 var isStream = require('is-stream');
 var Promise = require('pinkie-promise');
 
-var hasha = module.exports = function (buf, opts) {
+var hasha = module.exports = function (input, opts) {
 	opts = opts || {};
 
-	var inputEncoding = typeof buf === 'string' ? 'utf8' : undefined;
 	var outputEncoding = opts.encoding || 'hex';
 
 	if (outputEncoding === 'buffer') {
 		outputEncoding = undefined;
 	}
 
-	return crypto
-		.createHash(opts.algorithm || 'sha512')
-		.update(buf, inputEncoding)
-		.digest(outputEncoding);
+	var hash = crypto.createHash(opts.algorithm || 'sha512');
+
+	function update(buf) {
```

Comment:
Hmm, actually, I might be wrong. Seems that only applies to functions nested in a block, not function body.